GitUI/CommandsDialogs/SettingsDialog/Pages/BuildServerIntegrationSettingsPage.cs

```diff
@@ -56,9 +56,7 @@ protected override void SettingsToPage()
         {
             ThreadHelper.FileAndForget(async () =>
                 {
-                    Validates.NotNull(_populateBuildServerTypeTask);
-
-                    await _populateBuildServerTypeTask.JoinAsync();
+                    await Validates.NotNull(_populateBuildServerTypeTask).JoinAsync();
```

Comment:
Why are we checking `_populateBuildServerTypeTask` for null here?

Reply:
> Why are we checking `_populateBuildServerTypeTask` for null here?

One could answer "because a method is called".
But I guess you mean "Why was it necessary to move `NotNull`?". The analyzers complained about the not awaited `Task` returned by the standalone `NotNull`: https://ci.appveyor.com/project/gitextensions/gitextensions/builds/48731275#L99.
I still like the in-place `NotNull` much better.